qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -1,26 +1,42 @@
 #!/bin/sh -ex
 #
-# rbd_mirror_journal.sh - test rbd-mirror daemon in journal-based mirroring mode
+# rbd_mirror.sh - test rbd-mirror daemon in snapshot or journal mirroring mode
+#
+# Usage:
+#   RBD_MIRROR_MODE=journal rbd_mirror.sh
+#
+# Use environment variable RBD_MIRROR_MODE to set the mode
+# Available modes: snapshot | journal
 #
 # The scripts starts two ("local" and "remote") clusters using mstart.sh script,
 # creates a temporary directory, used for cluster configs, daemon logs, admin
 # socket, temporary files, and launches rbd-mirror daemon.
 #
 
-. $(dirname $0)/rbd_mirror_helpers.sh
+if [ "${#}" -gt 0 ]; then
+  echo "unnecessary arguments: ${@}"
+  exit 100
+fi
 
+if [ "${RBD_MIRROR_MODE}" != "snapshot" ] && [ "${RBD_MIRROR_MODE}" != "journal" ]; then
+  echo "unknown mode: ${RBD_MIRROR_MODE}"
+  echo "set RBD_MIRROR_MODE env variable, available modes: snapshot | journal"
+  exit 100
+fi
+
+. $(dirname $0)/rbd_mirror_helpers.sh
 setup
 
 testlog "TEST: add image and test replay"
 start_mirrors ${CLUSTER1}
 image=test
-create_image ${CLUSTER2} ${POOL} ${image}
+create_image_and_enable_mirror ${CLUSTER2} ${POOL} ${image} ${RBD_MIRROR_MODE}
 set_image_meta ${CLUSTER2} ${POOL} ${image} "key1" "value1"
 set_image_meta ${CLUSTER2} ${POOL} ${image} "key2" "value2"
 wait_for_image_replay_started ${CLUSTER1} ${POOL} ${image}
 write_image ${CLUSTER2} ${POOL} ${image} 100
 wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${image}
-wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${image} 'up+replaying' 'primary_position'
+wait_for_replay_status_in_pool_dir ${CLUSTER1} ${POOL} ${image} 'up+replaying'
```

Comment:
I think you don't need 'up+replaying' arg here. Isn't it always the same? That is why I suggested "wait_for_replay" name.

Reply:
Actually, I think a better name would be `wait_for_replaying_status_in_pool_dir`.

Reply:
I surely missed this, thanks!